kernel/src/scan/mod.rs

```diff
@@ -966,15 +951,18 @@ pub(crate) mod test_utils {
     ) {
         let logical_schema = logical_schema
             .unwrap_or_else(|| Arc::new(crate::schema::StructType::new_unchecked(vec![])));
+        let state_info = Arc::new(StateInfo {
+            logical_schema: logical_schema.clone(),
+            physical_schema: logical_schema.clone(),
```

Comment:
nit: you probably don't need this clone unless you use `logical_schema` again below